src/main/java/com/github/sergdelft/sqlcorgi/visitors/SelectStatementVisitor.java

```diff
@@ -225,7 +228,13 @@ private void handleHaving(PlainSelect plainSelect) {
      */
     private void handleJoins(PlainSelect plainSelect) {
         JoinRulesGenerator joinRulesGenerator = new JoinRulesGenerator();
-        Set<String> out = joinRulesGenerator.generate((PlainSelect) copy(plainSelect));
+        Set<String> out = joinRulesGenerator.generate((PlainSelect) copy(plainSelect), null);
+
+        if (joinRulesGenerator.getSanitized() != null) {
+            this.plainSelect = (PlainSelect) copy(joinRulesGenerator.getSanitized());
```

Comment:
Should we perhaps return the `PlainSelect` instead of storing it as a class member? There's no other place where we need to use it.

Reply:
That would work as well! It has been adjusted.